src/west/_bootstrap/main.py

```diff
@@ -115,26 +115,59 @@ def init(argv):
     This exits the program with a nonzero exit code if fatal errors occur.'''
     init_parser = argparse.ArgumentParser(
         prog='west init',
-        description='Bootstrap initialize a Zephyr installation')
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=
+'''
+Initializes a West instance.
+
+In more detail, does the following:
+
+  1. Clones the manifest repository to west/manifest, and the west repository
+     to west/west
+
+  2. Creates a marker file west/{}
+
+  3. Creates an initial configuration file west/config
+
+'west init' can be rerun on an already initialized West instance to update
+configuration settings. This is just an alternative to manually editing
+west/config. Only explicitly passed configuration values (e.g.
+--mr NEW_MANIFEST_REVISION) are updated.
+
+Note: If you update the west/manifest revision or URL, the change will take
+effect whenever the configuration setting is next read, e.g. in `west fetch`.
+'west init' only updates configuration settings.
```

Comment:
I think that running west init again should result in the same repository state for the manifest and west repositories as if you had run it for the first time whenever possible, rather than this kind of halfway change. I understand that this may not be possible due to a failed rebase if there are out of tree patches sometimes, but that seems like a separate issue.

Reply:
I think making it too magic would get messy. If you want to initialize another installation with completely different settings, then it's probably better to start from scratch.

Making e.g. `git clone` act as if you had cloned a different repository instead of erroring out wouldn't be the greatest design, to make a parallel. :)

Reply:
> Making e.g. `git clone` act as if you had cloned a different repository instead of erroring out wouldn't be the greatest design, to make a parallel. :)

The parallel here is with `repo init`, not `git clone`, I think.

> We could do it then and there as well, but it would be nice if there was a way to share code between the bootstrapper script and the rest of West before doing that, to avoid a bunch of code duplication for a minor thing like that.

I don't see much of an obstacle to doing that. Here are a couple alternatives:

1. put it in the bootstrapper and import the code from the rest of west (importing internal APIs should always be fine
2. make a `west.common` or so and add that package to what gets uploaded to pypi

Reply:
@mbolivar 
`west update` is now run automatically if you rerun `west init` to change settings. That means the special repositories get updated right away (if they can be fast-forwarded).

Do you think the self-update should be special-cased when you change settings via `west init`? What command should be run instead in that case? I guess the reasons we added the fast-forward-only behavior might still apply here.

Personally, I prefer the `repo` behavior (a plain and super simple `git rebase`, with a nice merge conflict that tells you what conflicts), but we already settled on the fast-forward-only behavior I think. `repo` so far has always worked just like I expected it to work. :)